networkx/algorithms/community/modularity_max.py

```diff
@@ -234,7 +215,98 @@ def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1)
             b[v] += b[u]
             b[u] = 0
 
-    return sorted(communities.values(), key=len, reverse=True)
+        yield list(communities.values())
```

Comment:
Another minor detail.  
Is it better to yield a list (causing many probably unneeded lists to be formed) or to yield the values object? 
If the caller wants a list they can make one. If they don't want a list, they can't change this line.